lib/chainsync/chainsync/analysis/calc_pnl.py

```diff
@@ -8,11 +8,14 @@
 from eth_typing import ChecksumAddress, HexAddress, HexStr
 from ethpy.base import smart_contract_preview_transaction
 from ethpy.hyperdrive import BASE_TOKEN_SYMBOL
+from ethpy.hyperdrive.interface import HyperdriveReadInterface
 from fixedpointmath import FixedPoint
 from web3.contract.contract import Contract
 
 
-def calc_single_closeout(position: pd.Series, contract: Contract, pool_info: pd.DataFrame, min_output: int) -> Decimal:
+def calc_single_closeout(
```

Comment:
FYI if speed is a concern, you should consider changing all of the `assert` statements in this function

1) don't use asserts -- they're slow; it's better to do `if condition; raise error`. Also, if we compile with optimizations on then [they don't get executed at all](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement:~:text=The%20current%20code%20generator%20emits%20no%20code%20for%20an%20assert%20statement%20when%20optimization%20is%20requested%20at%20compile%20time), so if they're important asserts we should be doing the if/raise pattern anyway.

2) Even after changing them, they should be more concise & only when necessary (for example, that `assert isinstance(maturity, int)` line is unnecessary).

Reply:
I remember doing this for typechecking, but I just removed them, and didn't get any errors